[Alexander Lee] iP

Varstak opened the pull request:
# KevinBot

> Talking to your best friend is sometimes all the therapy you need. 

KevinBot keeps you company and remembers your tasks so you don't have to. It is: 
- Fun and interactive
- Efficient

Steps to download this app are shown below: 
1. Visit the website through this link [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html)
2. Click on **Duke Project**
3. Click 'Download' 
4. Navigate to Duke.java
5. Click on the **Play** button beside the `main` code snippet: 
```java
    public static void main(String[] args) {
        new Duke().run();
    }
```
6. and you can start using KevinBot! :+1:

Features Included:
- [x] Creating to-do lists
- [x] Saving lists locally 
- [ ] Reminders (coming soon!)

Commit:
Add Gradle support

Varstak committed:
Level 1 done

Varstak committed:
Level 2 done

Varstak committed:
Level-3 done

Varstak committed:
Level 4 done. List bug fixed.

Varstak committed:
Text UI testing done. UI tidied up.

Varstak committed:
Exceptions Handled.

Varstak committed:
Level 6 done. List renamed. Minor bugs resolved.

Varstak committed:
Level-7 done

Varstak committed:
Level-7 done

Varstak committed:
Level-8 done.

Varstak committed:
Merge branch 'branch-level-8'

Varstak committed:
More-OOP done

Varstak committed:
Refactored duke and created duke package

Varstak committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Varstak committed:
Created JUnit tests.

Varstak committed:
Level-9 done.

Varstak committed:
Added documentation for Find methods.

Varstak committed:
Added JavaDocs

hingen commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+
+public class Deadline extends Task {
+    String deadlineTime;
```
I'm concern that some might interpret this variable to be a `LocalTime` object base on the name. Perhaps consider changing to `deadlineTimeString`

Varstak replied:
Thanks for the suggestion!

hingen commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+    public String toString() {
+        return "[D]" + super.toString() + "(" + deadlineTime + "hrs" + ")";
+    }
+    public String saveString() {
```
Some might interpret this as a method that saves to storage the info of this object as a string. Can consider `getSaveString()` instead.

hingen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    public static String parseDate(String str) {
```
I fear some might find this misleading as `parseDate` could be interpreted as parsing the string to get a `LocalDate` object. Maybe consider `formatDateString(String)`?

Varstak replied:
i think that's a good idea!

hingen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+    public static String[] parseTask(String input) {
+        String[] splitCommand = input.split(" ", 2);
+        String type = splitCommand[0];
+        if(type.equals("todo")) {
```
Is there a missing white-space between `if` and `(`?

hingen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+        return displayDate;
+    }
+
+    public static String[] parseTask(String input) {
```
I fear that this name might be misleading for some developers for the same reason stated in the above comment. Perhaps consider `extractArgsFromInput(String)`?

hingen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,68 @@
+        }
+    }
+
+    public void getTasks(File file) {
```
`get` might imply that the method returns something. Perhaps consider `loadTasks(File)`?

hingen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,68 @@
+                }
+                TaskList.getContent().add(currentTask);
+            }
+        } catch (FileNotFoundException e){
```
Is there a missing whitespace between `)` and `{`?

hingen commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,42 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String status() {
```
Maybe consider starting the name of this method with a verb.

hingen commented on src/main/java/duke/TaskHandler.java:
```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    public String eventHandler(String input) throws EmptyContentException {
```
Notice that there are quite a few locations where method name does start with a verb. Perhaps consider doing so?

hingen commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.List;
+
+public class TaskList {
+    private static List<Task> taskList;
```
I feel some might misinterpret `taskList` to be an instance of the TaskList class, perhaps consider `tasks`?

Varstak replied:
yup i think that will make it more clear!

hingen commented on src/main/java/duke/UIText.java:
```diff
@@ -0,0 +1,29 @@
+
+import java.util.Scanner;
+
+public class UIText {
```
Is there a reason for choosing the name `UIText` instead of `Ui`?

Varstak replied:
I wanted it to only store text outputs for the bot and separate them from the functions which handles the processes.

hingen reviewed (commented):
Just some nits to fix.

Varstak committed:
Updated code to match coding standards.

Varstak committed:
Conflicts resolved.

Varstak committed:
Conflicts Resolved

Varstak committed:
Removed tasks file

ShiJiaAo commented on src/main/java/duke/TaskHandler.java:
```diff
@@ -0,0 +1,170 @@
+            return "Nice! I've marked this task as undone: " + "\n" + (listIndex) + ". " + content.get(index).toString();
+        } else if(!content.get(index).isDone) {
+            return "This task is already marked undone. ";
+        } else {
```
I would make the argument that `else` statements after `return` statements are redundant. Perhaps you can consider removing the `else`s for better readability, leaving only the `if`s! (I actually lost style marks in cs1010 for putting an `else` block after an `if... return`)

ShiJiaAo commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String statusToString() {
+
+        return "[" + getStatusIcon() + "] ";
+    }
```
Perhaps you can consider removing the extra newline in this method! Same for some other methods in this class as well.

ShiJiaAo commented on src/main/java/duke/TaskHandler.java:
```diff
@@ -0,0 +1,170 @@
+        } else {
+            String taskContent = content.get(index).toString();
+            content.remove(index);
+            return "Noted. I've removed this task:" + "\n" + taskContent + "\n" +
+                    String.format("Now you have %d tasks in the list.", content.size()) + "\n";
```
I like how you split this long return statement into two lines!

ShiJiaAo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,75 @@
+
+        newFile.getParentFile().mkdirs();
+
+        if (! newFile.exists()) {
```
Is there an extra whitespace after the `!`?

ShiJiaAo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import java.io.*;
```
Spotted wildcard import! I believe our module does not allow this

ShiJiaAo reviewed (commented):
Very nice code overall! I like closely you stuck with the style conventions, and it made everything easy to read and understand. Good job Alex!

Varstak committed:
Delete bug fixed

Varstak committed:
Refactored execution code.

Varstak committed:
Level-10 done.

Varstak committed:
Assertions added.

Varstak committed:
Removed dead code.

Varstak committed:
Reminders functionality done.

Varstak committed:
Merge pull request #3 from Varstak/branch-A-Assertions

Assertions added.

Varstak committed:
Merge pull request #2 from Varstak/branch-A-Code-Quality

Removed dead code.

Varstak committed:
Merge branch 'BCD-Extension'

# Conflicts:
#	src/main/java/duke/TaskHandler.java

Varstak committed:
Update welcome message on launch

Varstak committed:
Update README.md

Varstak committed:
Update README.md

Varstak committed:
Fix minor bugs and included UI screenshot

Varstak committed:
Merge branch 'master' of https://github.com/Varstak/ip

Varstak committed:
Modify gradle file

Varstak committed:
Add JavaDocs

Methods are quite unclear after changing the UI from text to javafx.

Editing the javadocs help developers better understand the
transition to javafx by separating the javafx elements
from the system ui elements.

Varstak committed:
Edit README.
